src/rules.jl

```diff
@@ -121,8 +185,28 @@ end
 ```
 =#
 
+"""
+    frule(f, xs...)
+
+Apply the forward-mode differentiation rule to `f` with the given arguments `xs`,
+returning a tuple of `f(xs...)` and an [`AbstractRule`](@ref) object which can be
+called to evaluate the rule. If no forward-mode rule has been defined for `f`,
+`nothing` is returned.
+
+See also: [`rrule`](@ref)
+"""
```

Comment:
`frule` returns a tuple of `AbstractRule`s - one for each of the function's outputs. 

Similarly, `rrule` returns a tuple of `AbstractRule`s, one for each of the function's inputs.

If the returned tuple would only contain a single `AbstractRule`, the single `AbstractRule` may be returned directly (basically just matching Julia's `return` behavior).